sklearnex/tests/test_memory_usage.py

```diff
@@ -148,41 +133,69 @@ def gen_clsf_data(n_samples, n_features):
     )
 
 
-def split_train_inference(kf, x, y, estimator):
+def get_traced_memory(queue=None):
+    if _is_dpc_backend and queue and queue.sycl_device.is_gpu:
+        return _backend.get_used_memory(queue)
+    else:
+        return tracemalloc.get_traced_memory()[0]
+
+
+def take(x, index, axis=0, queue=None):
+    xp, array_api = get_namespace(x)
+    if array_api:
+        return xp.take(x, xp.asarray(index, device=queue), axis=axis)
+    else:
+        return x.take(index, axis=axis)
```

Comment:
Can you clarify what this function is doing

Reply:
when we use fancy indices in numpy like
```python
example = np.mgrid[:20]
index = [0,5,4,3,10]
print(example[index]) # uses fancy indexing
```
its actually replicating numpy.take.  However, for array api, take will only work if the index array is also located on the same device. So this special function is necessary so that array_api inputs (like dpctl tensors) will work properly with train_test_split.  Its likely we need to look into our version of that function for proper dpnp and dpctl support. Also a good candidate for array_api rollout.